esphomeRemote/esphomeRemoteLight.h

```diff
@@ -2,30 +2,219 @@
+/* Home Assistant entities can have these modes in color_mode/supported_color_mode.
+ * Depending on the mode the light supports we want to show sliders or not.
+ *
+ * |color_mode | supports brightness | supports color_temp |
+ * |---------- | ------------------- | ------------------- |
+ * | UNKNOWN   |          -          |          -          |
+ * | ONOFF     |          -          |          -          |
+ * | BRIGHTNESS|          x          |          -          |
+ * | COLOR_TEMP|          x          |          x          |
+ * | HS        |          x          |          -          |
+ * | RGB       |          x          |          -          |
+ * | RGBW      |          x          |          -          |
+ * | RGBWW     |          x          |          x          |
+ * | WHITE     |          x          |          -          |
+ * | XY        |          x          |          -          |
+ */
+enum ColorModeType{
+    unknown_type,
+    onoff_type,
+    brightness_type,
+    color_temp_type,
+    hs_type,
+    rgb_type,
+    rgbw_type,
+    rgbww_type,
+    white_type,
+    xy_type
+};
+
 class LightService: public CustomAPIDevice, public Component {
   public:
     LightService(std::string newFriendlyName, std::string newEntityId, DisplayUpdateInterface& newCallback) : friendlyName(newFriendlyName), entityId(newEntityId), display(newCallback) {
       onState = false;
       subscribe_homeassistant_state(&LightService::state_changed, newEntityId.c_str());
+      subscribe_homeassistant_state(&LightService::min_mireds_changed, newEntityId.c_str(),"min_mireds");
+      subscribe_homeassistant_state(&LightService::max_mireds_changed, newEntityId.c_str(),"max_mireds");
+      subscribe_homeassistant_state(&LightService::brightness_changed, newEntityId.c_str(),"brightness");
+      subscribe_homeassistant_state(&LightService::color_temp_changed, newEntityId.c_str(),"color_temp");
+      subscribe_homeassistant_state(&LightService::color_mode_changed, newEntityId.c_str(),"color_mode");
     }
     std::string friendlyName;
     std::string entityId;
     DisplayUpdateInterface& display;
+    int local_brightness = -1;
+    int local_color_temp = -1;
+    bool is_brightness_in_sync = false;
+    bool is_color_temp_in_sync = false;
+    ColorModeType color_mode = unknown_type;
     bool onState;
+    int min_mireds = 0;
+    int max_mireds = 0;
+
+    void decTemperature() {
+        // only send update to home assistant if
+        // we have received a confirmation for the last
+        // state change (state changed from ha)
+        if(!is_color_temp_in_sync){
+            return;
+        }
+        is_color_temp_in_sync = false;
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"color_temp", to_string(local_color_temp - id(inc_color_temperature_step))} ,

```

Comment:
can you indent this over so its like ```
        const std::map< std::string, std::string > data = {
            {"entity_id",entityId.c_str()},
            {"brightness", to_string(local_brightness - id(inc_brightness_step))} ,
        };
```
ill try to add linting to this project so these comments are automatic

Reply:
Oh yeah, no problem 👍🏻